packages/react-components/src/utils/transformAlarmThresholdToSynchroChartsThreshold.ts

```diff
@@ -0,0 +1,62 @@
+import { YAnnotation } from '@iot-app-kit/core';
+import { AlarmData } from '../hooks/useAlarms';
+import { IoTEventsToSynchroChartsComparisonOperator } from '@iot-app-kit/source-iotsitewise';
+import { COMPARATOR_MAP, COMPARISON_OPERATOR } from '../common/constants';
+
+const createSynchroChartsThreshold = ({
+  thresholdValue,
+  severity,
+  labelText,
+  comparisonOperator,
+}: {
+  thresholdValue: number;
+  severity: number | undefined;
+  labelText: string;
+  comparisonOperator: COMPARISON_OPERATOR;
+}): YAnnotation => {
+  return {
+    color: 'red',
+    value: thresholdValue,
+    severity: severity,
+    showValue: false,
+    label: {
+      text: labelText,
+      show: true,
+    },
+    comparisonOperator: comparisonOperator,
+  };
+};
+
+export const transformAlarmsToSynchroChartsThreshold = (
+  alarm: AlarmData
+): YAnnotation | undefined => {
+  const { models, inputProperty, thresholds, status } = alarm;
+  const modelsFound = models && models.length !== 0;
+  const thresholdsFound = thresholds && thresholds?.length !== 0;
+
+  if (!status.isSuccess) return undefined;
```

Comment:
Don't think we should be purposefully returning or setting `undefined` anywhere. Instead you can just wrap everything in the if statement `if (state.isSuccess)` and just not return anything otherwise.